### ISSUE TITLE
Enhance CA1305 analyzer to cover interpolated strings

### ISSUE THREAD

**Issue by Alex-Sob** [dotnet/roslyn-analyzers#6769]
### Describe the problem you are trying to solve

The existing `CA1305: Specify IFormatProvider` rule does no seem to cover interpolated strings. Currently compiler emits no warnings if a value that can be formatted is used inside an interpolation expression:

```cs
var formatted = $"Date: {DateTime.Now}";
```

### Describe suggestions on how to achieve the rule

The existing analyzer could be enhanced, so that it would report diagnostic if an object of a type containing `ToString(IFormatProvider)` or an object implementing `ISpanFormattable` (I'm aware that it's actually used instead of calling `ToString` on values) is injected into interpolated string. Does that make sense? If so, I could try contributing myself as I have experience writing analyzers for Microsoft projects.
